devito/data/data.py

```diff
@@ -260,108 +409,54 @@ def _convert_index(self, glb_idx):
 
         return loc_idx[0] if len(loc_idx) == 1 else tuple(loc_idx)
 
+    def _set_global_idx(self, val, idx, val_idx):
+        """
+        Compute the global indices to which val (the locally stored data) correspond.
+        """
+        data_loc_idx = as_tuple(val._index_glb_to_loc(val_idx))
+        data_global_idx = []
+        # Convert integers to slices so that shape dims are preserved
+        if is_integer(as_tuple(idx)[0]):
+            data_global_idx.append(slice(0, 1, 1))
+        for i, j in zip(data_loc_idx, val._decomposition):
+            if not j.loc_empty:
+                data_global_idx.append(j.index_loc_to_glb(i))
+            else:
+                data_global_idx.append(None)
+        mapped_idx = []
+        # Based data_global_idx the indices to which the locally stored data
```

Comment:
nitpicking (for homogeneity with rest of devito): use backtips for variable names in comments, ie \`data_global_idx\`

also, it should be "glb", not "global", again for consistency/homogeneity ?